hpc_launcher/schedulers/flux.py

```diff
@@ -82,11 +82,14 @@ def build_command_string_and_batch_script(
             header.write(f"# FLUX: {tmp}\n")
 
         # Set the Number of GPUs per task
+        # There is a difference in option names between tasks and allocations
         if self.gpus_per_proc > 0:
-            tmp = f"--gpus-per-task={self.gpus_per_proc}"
-            cmd_args += [tmp]
-            if not blocking:
-                header.write(f"#FLUX: {tmp}\n")
+            tmp = f"{self.gpus_per_proc}"
+            # command line flag for a task
+            self.run_launch_args["--gpus-per-task"] = tmp
+            # command and shell flags for an allocation
+            self.batch_submit_args["--gpus-per-slot"] = tmp
+            self.batch_script_header["# FLUX: --gpus-per-slot"] = tmp
```

Comment:
we should probably not set this

Reply:
I think that in a nested run we want it.  To be consistent with their expected behavior.  It should have no effect on primary allocations.  That is how I read it.

Reply:
FYI, Everett confirmed that the PR worked for him